src/Psalm/Internal/Analyzer/Statements/Expression/SimpleTypeInferer.php

```diff
@@ -92,11 +92,11 @@ public static function infer(
                         return Type::getString($result);
                     }
 
-                    if ($left->isSingle() && $left->getSingleAtomic() instanceof TNonEmptyString) {
+                    if ($left->isSingle() && $left->isNonEmptyString()) {
```

Comment:
Hey @danog! The isSingle check feels redundant. I think it was there for simplicity in order to be able to call getSingleAtomic just after but isNonEmptyString should be able to look through every Atomic

Reply:
Right, hang on let me push a fix